pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -95,6 +101,111 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
-	// Do nothing for now
+	for _, account := range userRecord.Spec.UserAccounts {
+		err = r.ensureUserAccount(reqLogger, account, userRecord)
+		if err != nil {
+			reqLogger.Error(err, "unable to synchronize with member UserAccount")
+			return reconcile.Result{}, err
```

Comment:
This comment is about brainstorming with team:

Should we add delay for next `Reconcile()` call with `RequeueAfter`??  Basically this `err` could be originated from member_cluster (remote call) and we may need to treat it differently ??  The probable error could be n/w issue and re-trying in short delay would not make sense ??

Reply:
This is a very big topic. In one of my versions of synchronization, I had different strategies (with different delays) for different types of error. In KubeFed they use something similar so I took the idea from them. The reasons why I haven't introduced it in our code yet are:
* I wanted to start with simple code and make it more complex later - so for now just use errors and default delays
* I think that it would be better to firstly know which types of errors we can face and based on this knowledge we can bring the strategies.

I'll create an Jira issue for this

Reply:
I agree. Let's keep it simple for now and improve later (if needed).
It seems that OpenShift (k8s?) already have some mechanism which increases delays between failed reconciles between attempts.
I noticed that if there is a constant failure then the reconcile is invoked with increasing delay (something like NextDelay = PreviousDelay * 2) and I think it's good enough for all our errors for now.